cocoa/tests_backend/window.py

```diff
@@ -22,10 +22,20 @@ def __init__(self, app, window):
         self.native = window._impl.native
         assert isinstance(self.native, NSWindow)
 
-    async def wait_for_window(self, message, minimize=False, full_screen=False):
+    async def wait_for_window(
+        self,
+        message,
+        minimize=False,
+        full_screen=False,
+        state_switch_not_from_normal=False,
+    ):
         await self.redraw(
             message,
-            delay=0.75 if full_screen else 0.5 if minimize else 0.1,
+            delay=(
+                1.75
+                if state_switch_not_from_normal
```

Comment:
The reason for a higher waiting delay in cases of state switching from a non-normal state to any another state is because to go from state A to state B, the window also passes through normal state i.e,
For switching from( `State A -> State B` ), the backend does( `State A -> NORMAL -> State B` ).

So the total waiting delay before we can assert the instantaneous state is:
`Delay of switching from(State A -> NORMAL)+ Delay of switching from(NORMAL -> State B)`